src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php

```diff
@@ -113,4 +113,21 @@ private function executeSearchQuery($expectedResults = 1)
 
         return $results;
     }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRename()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin');
+
+        $result = $this->executeSearchQuery(1);
+        $entry = $result[0];
+        $this->assertEquals($entry->getAttribute('cn')[0], 'Kevin');
```

Comment:
The old DN no longer exists if there's only 1 entry. If more than 1 Entry is found, the `assertCount` check will fail in `executeSearchQuery`, therefore I don't think it necessary to have another check.